css-view-transitions-2/Overview.bs

```diff
@@ -0,0 +1,373 @@
+
+''@auto-view-transition'' rules are [=CSS/parsed=] according to the following grammar,
+plus the additional rules noted below:
+
+<pre class=prod>
+	@auto-view-transition = @auto-view-transition { <<declaration-rule-list>> }
+</pre>
+
+## The [=@auto-view-transition/same-origin=] property ## {#view-transition-name-prop}
+
+	<pre class='descdef'>
+	Name: same-origin
+	For: @auto-view-transition
+	Value: enabled | disabled
+	Initial: disabled
+	</pre>
+
+	The '<dfn for="@auto-view-transition">same-origin</dfn>' property opts in to automatically performing a view transition when performing a [=same origin=] navigation.
+	It needs to be enabled both in the old document (when unloading) and in the new document (when ready to render).
+
+	<dl dfn-type=value dfn-for="same-origin">
+		: <dfn>disabled</dfn>
+		:: There will be no transition.
+
+		: <dfn>enabled</dfn>
+		:: The transition will be enabled if the navigation is same-origin, without cross-origin
+			redirects.
+	</dl>
+
+# API # {#api}
+
+## The <dfn interface>PageRevealEvent</dfn> ## {#reveal-event}
+
+Note: this should go in the HTML spec. See [Issue 9315](https://github.com/whatwg/html/issues/9315).
+
+<xmp class=idl>
+		[Exposed=Window]
+		interface PageRevealEvent : Event {
+			readonly attribute ViewTransition? viewTransition;
+		};
+</xmp>
+
+The <dfn attribute for=PageRevealEvent>viewTransition</dfn> [=getter steps=] are to return the
+[=inbound cross-document view-transition=] for [=this's=] [=relevant global object's=] [=associated document=].
+
+
+
+# Algorithms # {#algorithms}
+
+## Additions to {{ViewTransition}} ## {#view-transitions-extension}
+
+	A {{ViewTransition}} additionally has:
+	<dl dfn-for=ViewTransition>
+		: <dfn>is cross-document</dfn>
+		:: a boolean, initially false.
+	</dl>
+
+## Monkey patches to HTML ## {#monkey-patch-to-html}
+
+	<div algorithm="monkey patch to unload">
+		Run the following step before <a href="https://html.spec.whatwg.org/multipage/document-lifecycle.html#unloading-documents:update-the-visibility-state">updating the visibility state</a> in the [=unload=] steps:
+
+		1. If |newDocument| is given, then [=setup outbound cross-document view transition=] given |oldDocument| and |newDocument|.
+	</div>
+
+	<div algorithm="monkey patch to render blocking">
+		Run the following step at the end of [=unblock rendering=]:
+
+		1. [=reveal document|reveal=] |document|.
+
+		Reword the text in the [=render-blocked=] definition, to call [=reveal document=] |document| if the [=implementation-defined=] timeout value has been exceeded.
+	</div>
+
+	<div algorithm="monkey patch to reactivation">
+		Run the following step in [=Document/reactivate=], before step 4 (querying for [=page showing=]):
+
+		1. [=reveal document|reveal=] |document|.
+	</div>
+
+	<div algorithm="page reveal">
+	To <dfn>reveal {{Document}}</dfn> |document|:
+		1. Assert: |document|'s [=page showing=] is false.
+
+		1. If |document| is [=render-blocked=] then return.
```

Comment:
Why this check? We call this function when render blocking is finished (or timed out). So assert that we're not render-blocked?

Reply:
We also call it on reactivation, which could happen when the document is still render blocked (e.g. you activate a prerendered page really quickly before it's unblocked). To clarify, I moved this line into the reactivation bit.